File: api/info/info.go
package info

import "strings"

// Class represents which information classes are requested/subscribed
type Class uint8

const (
	Status Class = 1 << iota // Node status information
	Config                   // Node configuration
	Logs                     // Node logs
)

// String returns a human-readable representation of Class for debugging
func (c Class) String() string {
	if c == 0 {
		return "None"
	}

	var parts []string
	if c&Status != 0 {
		parts = append(parts, "Status")
	}
	if c&Config != 0 {
		parts = append(parts, "Config")
	}
	if c&Logs != 0 {
		parts = append(parts, "Logs")
	}
	return strings.Join(parts, "|")
}

File: web/src/info.ts
// Info Class Bitmask
// Must match api/info/info.go

export const enum info {
  Status = 1 << 0, // Node status information
  Config = 1 << 1, // Node configuration
  Logs = 1 << 2, // Node logs
}

// toString returns a human-readable representation for debugging
export function infoToString(c: number): string {
  if (c === 0) {
    return "None"
  }

  const parts: string[] = []
  if (c & info.Status) parts.push("Status")
  if (c & info.Config) parts.push("Config")
  if (c & info.Logs) parts.push("Logs")
  return parts.join("|")
}
